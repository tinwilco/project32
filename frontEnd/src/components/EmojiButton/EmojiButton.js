import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames/bind";

import styles from "./EmojiButton.module.css";

const cx = classnames.bind(styles);

const EmojiButton = ({
  label,
  handleButtonPress,
  disabled,
  color,
  selected
}) => {
  return (
    <button
      className={cx({ EmojiButton: true, EmojiButton_selected: selected })}
      type="button"
      onClick={handleButtonPress}
      disabled={disabled}
    >
      {label}
    </button>
  );
};

EmojiButton.propTypes = {
  label: PropTypes.string.isRequired,
  handleButtonPress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  color: PropTypes.string,
  selected: PropTypes.bool
};

EmojiButton.defaultProps = {
  disabled: false,
  selected: false,
  color: "lightgrey"
};

export default EmojiButton;
