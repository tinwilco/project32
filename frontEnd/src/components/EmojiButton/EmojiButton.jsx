import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames/bind";

import styles from "./EmojiButton.module.css";

const cx = classnames.bind(styles);

const EmojiButton = ({
  label,
  emoji,
  handleButtonPress,
  disabled,
  selected
}) => (
  <button
    className={cx({ EmojiButton: true, EmojiButton_selected: selected })}
    type="button"
    onClick={() => handleButtonPress(label)}
    disabled={disabled}
  >
    <span role="img" aria-label={label}>
      {emoji}
    </span>
  </button>
);

EmojiButton.propTypes = {
  label: PropTypes.string.isRequired,
  handleButtonPress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  emoji: PropTypes.string.isRequired,
  selected: PropTypes.bool
};

EmojiButton.defaultProps = {
  disabled: false,
  selected: false
};

export default EmojiButton;
