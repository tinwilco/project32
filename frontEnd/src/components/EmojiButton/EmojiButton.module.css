.EmojiButton {
  height: 100px;
  width: 100px;
  margin: 20px;
  background-color: lightgray;
  font-size: 48px;
  border-radius: 100px;
  border-width: 1px;
  border-color: black;
}

.EmojiButton_selected {
  background-color: blueviolet;
}
