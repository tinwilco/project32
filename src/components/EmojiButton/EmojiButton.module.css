@import "../../assets/colors.css";

.EmojiButton {
  height: 100px;
  width: 100px;
  margin: 20px;
  background-color: var(--button);
  font-size: 48px;
  border-radius: 100px;
  border-width: 1px;
  border-color: var(--border-dark);
  transition: background-color ease-in-out 0.3s;
}

.EmojiButton_selected {
  background-color: var(--highlight);
}
