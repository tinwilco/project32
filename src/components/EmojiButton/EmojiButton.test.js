/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { render, screen } from "@testing-library/react";

import userEvent from "@testing-library/user-event";

import EmojiButton from "./EmojiButton";

const defaultProps = {
  label: "test",
  emoji: "ðŸ§ª",
  handleButtonPress: jest.fn(),
  disabled: false,
  selected: false
};

describe("EmojiButton component", () => {
  it("renders to match the snapshot", () => {
    const subject = render(<EmojiButton {...defaultProps} />);
    expect(subject.baseElement).toMatchSnapshot();
  });
  it("correctly calls the handleButtonPress method when the button is clicked", () => {
    render(<EmojiButton {...defaultProps} />);
    expect(defaultProps.handleButtonPress).not.toHaveBeenCalled();
    userEvent.click(screen.getByText(defaultProps.emoji).closest("button"));
    expect(defaultProps.handleButtonPress.mock.calls[0][0]).toBe(
      defaultProps.label
    );
  });
});
