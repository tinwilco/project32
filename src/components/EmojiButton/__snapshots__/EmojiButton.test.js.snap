// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmojiButton component renders to match the snapshot 1`] = `
<body>
  <div>
    <button
      class="EmojiButton"
      type="button"
    >
      <span
        aria-label="test"
        role="img"
      >
        ðŸ§ª
      </span>
    </button>
  </div>
</body>
`;
